x-pack/plugins/elastic_assistant/server/routes/post_actions_connector_execute.ts

```diff
@@ -144,11 +146,25 @@ export const postActionsConnectorExecuteRoute = (
           if (onLlmResponse) {
             await onLlmResponse(error.message, {}, true);
           }
+          const pluginName = getPluginNameFromRequest({
+            request,
+            defaultPluginName: DEFAULT_PLUGIN_NAME,
+            logger,
+          });
+          const v2KnowledgeBaseEnabled =
+            assistantContext.getRegisteredFeatures(pluginName).assistantKnowledgeBaseByDefault;
+          const kbDataClient =
+            (await assistantContext.getAIAssistantKnowledgeBaseDataClient(
+              v2KnowledgeBaseEnabled
+            )) ?? undefined;
```

Comment:
So https://github.com/elastic/kibana/pull/186566 was (intentionally) not backported to `8.15`, so there is no concept of `v2KnowledgeBaseEnabled` when fetching the `kbDataClient` on that branch. That said, all the other changes look to be compatible, so when backporting, this portion can just be updated to not provide that argument and we should be good here 👍